python/taichi/lang/simt/warp.py

```diff
@@ -28,15 +28,24 @@ def shfl_i32(mask, val, offset):
 
 
 def shfl_down_i32(mask, val, offset):
+    # we use 31 as the last argument since 32 (warp size) does not work
```

Comment:
```suggestion
    # Here we use 31 as the last argument since 32 (warp size) does not work
    # for some reason. Using 31 leads to the desired behavior.
```